src/generic/grid.cpp

```diff
@@ -6889,8 +6893,11 @@ void wxGrid::DrawColLabel(wxDC& dc, int col)
     {
         // It is reported that we need to erase the background to avoid display
         // artefacts, see #12055.
-        wxDCBrushChanger setBrush(dc, m_colLabelWin->GetBackgroundColour());
-        dc.DrawRectangle(rect);
+        {
+            wxDCBrushChanger setBrush(dc, m_colLabelWin->GetBackgroundColour());
+            wxDCPenChanger setPen(dc, m_colLabelWin->GetBackgroundColour());
```

Comment:
It would be slightly better to use `wxTRANSPARENT_PEN` here, it should have the same effect, but is supposed to be more efficient.